### PR TITLE
win_service_stat: Added module

### PR DESCRIPTION
**Pull request by jborean93**
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
win_service_stat

##### ANSIBLE VERSION
```
ansible 2.3.0 (devel 98934939af) last updated 2017/02/25 13:36:22 (GMT +1000)
  config file = /mnt/c/dev/test/role/ansible.cfg
  configured module search path = [u'/mnc/c/dev/test/role/library']
```

##### SUMMARY
Added new Windows module called win_service_stat to get information about a windows service and if it exists or not. Was going to include this in win_service module but I feel that it will make it too complex and go against the common practice where we have an action module like win_file or win_regedit and a stat module like win_stat or win_reg_stat.

### CODE REVIEW

**Comment by jborean93** [comment]
@jhawkesworth @dagwieers @nitzmahone After having a think I feel like having a module the checks if a service exists as well as returning info about that service should be in its own module like we have done with win_stat and others. I am hoping to get this merged in quickly so I can finish off the tests properly for https://github.com/ansible/ansible/pull/19143 which I originally put the functionality in.

File: lib/ansible/modules/windows/win_service_stat.ps1 [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params "_ansible_check_mode" -type "bool" -default $false
```
Comment (dagwieers) [inline lib/ansible/modules/windows/win_service_stat.ps1:23]: -name is missing (not strictly needed, but we do it for everything else)

File: lib/ansible/modules/windows/win_service_stat.ps1 [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    $actual_start_mode = $wmi_svc.StartMode.ToString().ToLower() 
+    if ($delayed -and $actual_start_mode -eq 'auto') {
+        $actual_start_mode = 'delayed'
+    }
+
+    $existing_depenencies = @()
+    $existing_depended_by = @()
+    if ($svc.ServicesDependedOn.Count -gt 0) {
+        foreach ($dependency in $svc.ServicesDependedOn.Name) {
+            $existing_depenencies += $dependency
```
Comment (dagwieers) [inline lib/ansible/modules/windows/win_service_stat.ps1:51]: I see a typo in the variable name: **$existing_depenencies**

**Review by dagwieers — CHANGES_REQUESTED** [review]

**Comment by dagwieers** [comment]
The module itself looks fine. Although I am not sure this is what the project wants to do.
Your module is not the only module, I opened a similar question for the next core meeting related to this.
https://github.com/ansible/community/issues/150#issuecomment-282733423

**Comment by jborean93** [comment]
Thanks @nitzmahone but if you merged in the win_service changes we may want to merge this out as the functionality is in there.

**Comment by nitzmahone** [comment]
Ugh, clearly I'm getting punchy. Yeah, if the functionality is completely superseded by the stuff in win_service, I'll just revert this one.

**Comment by jborean93** [comment]
No worries I'll have a look at the other modules you merged in and try and make the changes that were requested. Minor things like documentation and syntax changes are still fine for now right?

**Comment by nitzmahone** [comment]
Absolutely- the intent is just to freeze functionality to give it some time to settle (so no new modules, no new options on existing modules, no major behavior changes). Cleaning up stuff that's already there is fine.

**Comment by jborean93** [comment]
Awesome, a lot of my spare time is taken up with psrp I'm trying out but am hoping to get to it soon.

**Comment by nitzmahone** [comment]
I'm curious if you actually find any benefit to PSRP (for Ansible)- with the relatively chunky way we talk to Powershell, it seems like PSRP is a much chattier and verbose way of accomplishing roughly the same thing. Now that Nano Server (secretly) supports `-EncodedCommand`, I don't see any *need* for PSRP anywhere in Ansible, but let me know if you have a different take.

**Comment by jborean93** [comment]
I definitely will, this is probably more for my benefit to understand the protocol more and just do something different but it is still early days.

**Pull request merged**